packages/gatsby/src/schema/node-model.js

```diff
@@ -341,13 +342,10 @@ class LocalNodeModel {
    * @param {Node} node Root Node
    */
   trackInlineObjectsInRootNode(node) {
-    return addRootNodeToInlineObject(
-      this._rootNodeMap,
-      node,
-      node.id,
-      true,
-      true
-    )
+    if (!this._trackedRootNodes.has(node.id)) {
```

Comment:
does this work with hot-reloading, i.e. when the node.id stays the same but some nested field changes and needs to be re-tracked?

Reply:
Changing nodes resets node model.